### PR TITLE
codeintel: Add num_parts and uploaded_parts to lsif_uploads table

### PR DESCRIPTION
**Pull request by efritz**
This is a step to simplify the effort in https://github.com/sourcegraph/sourcegraph/pull/10456, which implements [RFC 143: Multipart uploads](https://docs.google.com/document/d/1nN9mKnQ-5OSPnWg0mHAJ9T3p1jEdnDXgb3HgcnHNckQ).

This adds two new fields to the `lsif_uploads` table: `num_parts` and `uploaded_parts` and adds the `uploaded` value to the LSIF state enum. The first field is the number of expected parts to be uploaded and the later is the index of parts that have already been uploaded. Once the length of the array is the number of expected parts, the parts can be _stitched_ together into a single upload on the backend.

The current behavior is to populate these fields with `1` and `[0]`, as multipart uploads are not yet implemented. These values are correct for uploads that have a single payload.

### CODE REVIEW

File: cmd/precise-code-intel/src/shared/models/pg.ts [hunk 1]
```diff
@@ -39,6 +39,14 @@ export class LsifUpload {
     @Column('text')
     public indexer!: string
 
+    /** The type of indexer used to produce the dump. */
+    @Column('int', { name: 'num_parts' })
+    public numParts!: number
+
+    /** The type of indexer used to produce the dump. */
```
Comment (gbrik) [inline cmd/precise-code-intel/src/shared/models/pg.ts:46]: reminder:
fix these descriptions

Comment (efritz) [reply cmd/precise-code-intel/src/shared/models/pg.ts:46]: Done.

**Comment by efritz** [comment]
> right now the fields are populated with a default but otherwise totally ignored?

Yes, and the defaults are actually correct once they will be read for single-payload uploads. They'll only have different values if the src-cli client states there will be more than one payload.

**Comment by gbrik** [comment]
How is looking up the upload (part) going to work/how does it work now?

**Comment by efritz** [comment]
> How is looking up the upload (part) going to work/how does it work now?

On upload of a single payload we just insert the upload record as 'queued' and it can be processed by the worker immediately.

On a multipart upload, we insert the upload record as 'uploading' so the worker doesn't fetch it. Subsequent requests will send part _i_ of the upload which is stored on disk (with the bundle id and the part index). Once all parts have been sent it will issue one last request that causes the parts to be concatenated into a single file, then the upload record is updated to 'queued' so that the worker can process it.

The index of each part written to disk is kept in the `uploaded_parts` column, and the number of total parts expected is the `num_parts` column. We use these numbers to validate out-of-range indexes and whether something is missing.

**Comment by gbrik** [comment]
got it, makes sense

**Pull request merged**